src/EventStore.Core/Authorization/LegacyAuthorizationProviderFactory.cs

```diff
@@ -19,29 +21,40 @@ public class LegacyAuthorizationProviderFactory : IAuthorizationProviderFactory
 			new Claim(ClaimTypes.Role, SystemRoles.Operations), new Claim(ClaimTypes.Name, SystemUsers.Operations)
 		};
 
-		public LegacyAuthorizationProviderFactory(IPublisher mainQueue) {
+		public LegacyAuthorizationProviderFactory(IPublisher mainQueue, bool allowAnonymousEndpointAccess, bool allowAnonymousStreamAccess) {
 			_mainQueue = mainQueue;
+			_allowAnonymousEndpointAccess = allowAnonymousEndpointAccess;
+			_allowAnonymousStreamAccess = allowAnonymousStreamAccess;
 		}
 
 		public IAuthorizationProvider Build() {
 			var policy = new Policy("Legacy", 1, DateTimeOffset.MinValue);
-			var streamAssertion = new LegacyStreamPermissionAssertion(_mainQueue);
+			var legacyStreamAssertion = new LegacyStreamPermissionAssertion(_mainQueue);
 
-			policy.AllowAnonymous(Operations.Node.Redirect);
-			policy.AllowAnonymous(Operations.Node.StaticContent);
 			policy.AllowAnonymous(Operations.Node.Ping);
-			policy.AllowAnonymous(Operations.Node.Options);
-
+			policy.AllowAnonymous(Operations.Node.Gossip.Read);
```

Comment:
`Operations.Node.Gossip.ClientRead` should be used for gossip requests from external clients, and `Operations.Node.Gossip.Read` should be used for node-to-node gossip.

As such, I think only `Gossip.ClientRead` should allow anonymous access. And perhaps the `Gossip.Read` operation should be further locked down to requiring system access?

That being said, the HTTP GossipController is registering `/gossip` as a `Gossip.Read` operation. This should really be changed to `Gossip.ClientRead` ([see here](https://github.com/EventStore/EventStore/blob/master/src/EventStore.Core/Services/Transport/Http/Controllers/GossipController.cs#L37))

Reply:
In summary, changes needed:
- `Gossip.ClientRead` should use the `allow-anonymous-endpoints` option
- `Gossip.Read` should require a system user
- `/gossip` in the HTTP gossipController should be changed to use `Gossip.ClientRead`